eZ/Publish/Core/Repository/Repository.php

```diff
@@ -260,21 +269,28 @@ public function __construct(
             $this->serviceSettings['language']['languages'] = $this->serviceSettings['languages'];
         }
 
-        if ($user !== null) {
-            $this->setCurrentUser($user);
+        if ($user instanceof User) {
+            $this->currentUser = $user;
+            $this->currentUserRef = new UserReference($user->getUserId());
```

Comment:
I meant this, basically:

```php
if ($user instanceof User) {
    $this->currentUser = $user;
    $this->currentUserRef = $user;
}
```

But maybe we should skip if it isn't that obvious :)